rwengine/src/ai/CharacterController.cpp

```diff
@@ -688,9 +688,37 @@ bool Activities::UseItem::update(CharacterObject *character,
             character->playCycle(shootcycle);
         }
     } else if (weapon->fireType == WeaponData::MELEE) {
-        RW_CHECK(weapon->fireType != WeaponData::MELEE,
-                 "Melee attacks not implemented");
-        return true;
+        auto currentAnim = character->getCurrentCycle();
```

Comment:
nit: `currentAnimCycle`?
nit2: Btw, this file is quite (too) complicated.

Reply:
I agree.